pulpcore/app/models/progress.py

```diff
@@ -142,6 +145,16 @@ def save(self, *args, **kwargs):
             super().save(*args, **kwargs)
             self._last_save_time = now
 
+    async def asave(self, *args, **kwargs):
+        """
+        Async save method that wraps the save() call with sync_to_async
+        args (list): positional arguments to be passed on to the real save
+        kwargs (dict): keyword arguments to be passed on to the real save
+        """
+        if not self.task_id:
+            self.task = await sync_to_async(Task.current)()
+        await sync_to_async(self.save)(*args, **kwargs)
```

Comment:
The `Task.current()` will be called in `save` anyway. So you might get away with:
```suggestion
    asave = sync_to_async(self.save)
```